### PR TITLE
Consolidate selection prompts into helper

### PR DESCRIPTION
**Pull request by gschier**
This PR consolidates the prompt selection logic into two helper `selectString` and `selectCustom`.

- All selections now support filtering and show 10 items
- All selections now include a caret for the active item

CleanShot 2021-06-11 at 09 54 51

### CODE REVIEW

File: cmd/add.go [hunk 1]
```diff
@@ -17,13 +17,18 @@ func (h *Handler) Add(ctx context.Context, req *entity.CommandRequest) error {
 	if err != nil {
 		return err
 	}
-	selection, err := ui.PromptPlugins(plugins)
+	selectedPlugin, err := ui.PromptPlugins(plugins)
 	if err != nil {
 		return err
 	}
+	ui.StartSpinner(&ui.SpinnerCfg{
```
Comment (gschier) [inline cmd/add.go:24]: Bonus tweak!

File: ui/prompt.go [hunk 2]
```diff
@@ -54,24 +50,21 @@ func hasTeams(projects []*entity.Project) bool {
...
-			if project.Team != nil {
-				tm[*project.Team] = true
+			if project.Team == nil {
+				continue
+			}
+
+			// Ensure teams are only appended once by checking teamCheck
+			if _, hasSeenTeam := teamCheck[*project.Team]; !hasSeenTeam {
+				teams = append(teams, *project.Team)
+				teamCheck[*project.Team] = true
```
Comment (coffee-cup) [inline ui/prompt.go:63]: I'm surprised Go doesn't have an Object.values like way to convert map to list of values

File: ui/prompt.go [hunk 3]
```diff
@@ -282,3 +236,59 @@ func validatorRequired(errorMsg string) func(s string) error {
...
+			index:     i,
+		})
+	}
+
+	options := &promptui.Select{
+		Label: fmt.Sprintf("Select %s", label),
+		Items: wrappedItems,
+		Size:  10,
+		Templates: &promptui.SelectTemplates{
+			Active:   fmt.Sprintf(`%s {{ . | underline }}`, promptui.IconSelect),
```
Comment (coffee-cup) [inline ui/prompt.go:281]: Is the custom triangle caret required here?

**Review by coffee-cup — APPROVED** [review]
Can't wait for this 😍

File: ui/prompt.go (hunk 2 above)
Comment (JakeCooper) [reply ui/prompt.go:63]: I am not at all surprised by this

https://twitter.com/JustJake/status/1395590075824541697

File: ui/prompt.go (hunk 3 above)
Comment (JakeCooper) [reply ui/prompt.go:281]: I think it's covered by the promptui.IconSelect element

Comment (coffee-cup) [reply ui/prompt.go:281]: oh 🤦  I missed that. Carry on!

File: ui/prompt.go (hunk 2 above)
Comment (gschier) [reply ui/prompt.go:63]: Once Generics are in, this should be possible

**Pull request merged**